src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder.cs

```diff
@@ -974,16 +974,24 @@ public DecisionDag(DagState rootNode)
             /// <summary>
             /// A successor function used to topologically sort the DagState set.
             /// </summary>
-            private static IEnumerable<DagState> Successor(DagState state)
+            private static ImmutableArray<DagState> Successor(DagState state)
             {
-                if (state.TrueBranch != null)
+
+                if (state.TrueBranch != null && state.FalseBranch != null)
                 {
-                    yield return state.TrueBranch;
+                    return ImmutableArray.Create(state.TrueBranch, state.FalseBranch);
```

Comment:
Is this done often enough to be wornth caching?

Reply:
No, because every caller of the topological sort caches the final result of the sort.

---
In reply to: [180259757](https://github.com/dotnet/roslyn/pull/26052#discussion_r180259757) [](ancestors = 180259757)

Reply:
I'm thinking about allocations. Caching the allocation of the array so that we keep these for other sorts in the future.